### PR TITLE
XW-3296 | new dropdowns

### PR DESCRIPTION
**Pull request by latata**
Don't merge before releasing DS

https://wikia-inc.atlassian.net/browse/XW-3296

@Wikia/x-wing

### CODE REVIEW

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/643089434cbb71ad82dd9b75571589c3bc837da5/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/643089434cbb71ad82dd9b75571589c3bc837da5/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/DesignSystem/bower.json [hunk 1]
```diff
@@ -3,6 +3,6 @@
   "private": true,
 
   "dependencies": {
-    "design-system": "Wikia/design-system#6.0.2"
+    "design-system": "Wikia/design-system#XW-3296"
```
Comment (rybmat) [inline extensions/wikia/DesignSystem/bower.json]: remember to change it to latest release

**Review by rybmat — CHANGES_REQUESTED** [review]

File: extensions/wikia/DesignSystem/styles/GlobalNavigationNotifications.scss [hunk 2]
```diff
@@ -97,6 +97,7 @@ $white: #fff;
 #GlobalNavigationWallNotifications {
 	background-color: $white;
 	border: 1px solid $border-color;
+	border-radius: 4px;
```
Comment (weronika) [inline extensions/wikia/DesignSystem/styles/GlobalNavigationNotifications.scss]: wds variable should be used here

**Review by weronika — CHANGES_REQUESTED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/3a5424ee9fc52ff1034def85d66f531863df9d6c/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/3a5424ee9fc52ff1034def85d66f531863df9d6c/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/08c5b4275256c37ab0484546d384840aa6946a33/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/08c5b4275256c37ab0484546d384840aa6946a33/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/1307da480d4eca7a2f738d08293954f33daa7c05/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/1307da480d4eca7a2f738d08293954f33daa7c05/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/d063e81c778da0600d7008a87f04433eb31c4f3a/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/d063e81c778da0600d7008a87f04433eb31c4f3a/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/7f2a71622fe7daff0d882977b03eb5bd4c81a1a5/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/7f2a71622fe7daff0d882977b03eb5bd4c81a1a5/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/f455b96dad80548ef5e935255bd1cf6dbfd4d4db/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/f455b96dad80548ef5e935255bd1cf6dbfd4d4db/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/1787b0208baf52fa8227456a8da20fe5a3673e42/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/1787b0208baf52fa8227456a8da20fe5a3673e42/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/10b57168a6b9966f0782234a3c817f81ee65636b/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/10b57168a6b9966f0782234a3c817f81ee65636b/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/8e85f409cae4c8129e813b43fec70d198d960151/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/8e85f409cae4c8129e813b43fec70d198d960151/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/50aab3c4f788692a6e96c76c1cbc0fd39cb7a71a/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/50aab3c4f788692a6e96c76c1cbc0fd39cb7a71a/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js [hunk 3]
```diff
@@ -42,9 +42,9 @@ require(
 				this.unreadCount = parseInt(this.$notificationsCount.html(), 10);
 
 				this.$notificationsEntryPoint
-					.on('click', this.proxy(this.updateCounts))
-					.on('click', this.proxy(this.fetchForCurrentWiki))
-					.on('wds-dropdown-open', this.proxy(this.onNotificationsOpen));
+					.on('mouseenter', this.proxy(this.updateCounts))
```
Comment (hakubo) [inline extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js:45]: good catch!

**Review by hakubo — APPROVED** [review]

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/72a1614ac66cfa5255d6c61c9f9d99222efd0848/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/72a1614ac66cfa5255d6c61c9f9d99222efd0848/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

File: extensions/wikia/DesignSystem/scripts/DesignSystemDropdowns.js [hunk 4]
```diff
@@ -0,0 +1,19 @@
+$(function ($) {
+	'use strict';
+
+	var className = 'is-clicked';
+
+	// On touch devices, the first click opens dropdown, the second one opens the link if there is any
+	if (window.Wikia.isTouchScreen()) {
+		$('body')
+			.on('touchstart', '.wds-dropdown', function (e) {
```
Comment (hakubo) [inline extensions/wikia/DesignSystem/scripts/DesignSystemDropdowns.js]: please call it 'event' for readability

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/da5b381ea09e65769058f5650ff69b0253cced6b/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/da5b381ea09e65769058f5650ff69b0253cced6b/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Comment by WikiaTech** [comment]
SonarQube analysis reported 2 issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationNotifications.js#L52](https://github.com/Wikia/app/blob/36faf68b39d35e80ed7eb3094c2f1299e31e2cb1/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationNotifications.js#L52): Remove the unused function parameter "ev". [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AUnusedFunctionArgument)
1. ![MAJOR][MAJOR] [DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10](https://github.com/Wikia/app/blob/36faf68b39d35e80ed7eb3094c2f1299e31e2cb1/extensions/wikia/DesignSystem/scripts/DesignSystemGlobalNavigationOnSiteNotifications.view.js#L10): Function has 8 parameters which is greater than 4 authorized. [rule](https://sonarqube.wikia-inc.com/coding_rules#rule_key=javascript%3AExcessiveParameterList)

[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'

**Pull request merged**